cmd/empire/factories.go

```diff
@@ -269,7 +269,9 @@ func newHBReporter(key, env string) (reporter.Reporter, error) {
 
 func newAuthProvider(c *cli.Context) (dockerauth.AuthProvider, error) {
 	provider := dockerauth.NewMultiAuthProvider()
-	provider.AddProvider(dockerauth.NewECRAuthProvider(ecr.New(newConfigProvider(c))))
+	provider.AddProvider(dockerauth.NewECRAuthProvider(func(region string) dockerauth.ECR {
+		return ecr.New(newConfigProvider(c), &aws.Config{Region: aws.String(region)})
```

Comment:
If wonder if by doing memoization here you introduce thread-safety concerns. According to the docs, `http.Serve` runs each request in its own goroutine, and I don't think Go makes any atomicity guarantees in most cases.

I did a bit of benchmarking on session creation vs copying:

```go
func BenchmarkAwsSessionCreate(b *testing.B) {
  var svc *ecr.ECR

  for i := 0; i < b.N; i++ {
    svc = ecr.New(session.New(), &aws.Config{Region: aws.String("us-east-1")})
  }

  _ = svc
}

func BenchmarkAwsSessionCopy(b *testing.B) {
  sess := session.New()
  var svc *ecr.ECR

  for i := 0; i < b.N; i++ {
    svc = ecr.New(sess, &aws.Config{Region: aws.String("us-east-1")})
  }

  _ = svc
}
```

```
BenchmarkAwsSessionCreate-8	  200000	      9235 ns/op
BenchmarkAwsSessionCopy-8  	  300000	      5401 ns/op
```

So I think I'll move it out for now.